src/main/java/athena/task/Task.java

```diff
@@ -71,22 +75,89 @@ public Task(String name, String startTime, String duration, String deadline,
 
 
 
+        setRecurrence(recurrence);
+    }
+
+
+    private void setRecurrence(String recurrence) {
         //TODO: refactor this into Time class
 
         if (recurrence.toUpperCase().equals(Recurrence.TODAY.toString())) {
-            recurrenceDate = LocalDate.now();
+            recurrenceDates.add(LocalDate.now());
+        } else if (recurrence.toUpperCase().equals(Recurrence.MONDAY.toString())) {
+            LocalDate mondayDate = getFirstDay(DayOfWeek.MONDAY);
+            addDates(mondayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.TUESDAY.toString())) {
+            LocalDate tuesdayDate = getFirstDay(DayOfWeek.TUESDAY);
+            addDates(tuesdayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.WEDNESDAY.toString())) {
+            LocalDate wednesdayDate = getFirstDay(DayOfWeek.WEDNESDAY);
+            addDates(wednesdayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.THURSDAY.toString())) {
+            LocalDate thursdayDate = getFirstDay(DayOfWeek.THURSDAY);
+            addDates(thursdayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.FRIDAY.toString())) {
+            LocalDate fridayDate = getFirstDay(DayOfWeek.FRIDAY);
+            addDates(fridayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.SATURDAY.toString())) {
+            LocalDate saturdayDate = getFirstDay(DayOfWeek.SATURDAY);
+            addDates(saturdayDate);
+        } else if (recurrence.toUpperCase().equals(Recurrence.SUNDAY.toString())) {
+            LocalDate sundayDate = getFirstDay(DayOfWeek.SUNDAY);
+            addDates(sundayDate);
         } else {
             try {
-                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
-                recurrenceDate = LocalDate.parse(recurrence, formatter);
+                setRecurrenceDate(recurrence);
             } catch (DateTimeParseException e) {
                 // TODO: Handle this properly
                 System.out.println("I don't understand the date you gave. So I set it to today.");
-                recurrenceDate = LocalDate.now();
+                recurrenceDates.add(LocalDate.now());
             }
         }
     }
 
+    private void addDates(LocalDate startDate) {
+        for (int i = 0; i < 10; i++) { // Max number of weeks now is 10 to prevent infinite recurrence
+            recurrenceDates.add(startDate.plusWeeks(i));
+        }
+    }
+
+    private LocalDate getFirstDay(DayOfWeek dayOfWeek) {
+        LocalDate startDate = LocalDate.now();
+        for (int i = 1; i < 7; i++) {
```

Comment:
small nitpick, i can start from 0 then go to i<6, functionality wise is the same but might be more readable for others